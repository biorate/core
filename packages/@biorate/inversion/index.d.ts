import { Container } from 'inversify';
import { IService } from './interfaces';
import { IDecorator } from '@biorate/lifecycled';
export * from './interfaces';

declare module '@biorate/inversion' {
  export { init, kill, on } from '@biorate/lifecycled';

  export function Core<T extends new (...any: any[]) => any>(Class?: T): T;

  export function injectable(): IDecorator;

  export function factory(
    Type: Object,
    Child: Object,
    Parent: Object,
    Root: Object,
  ): void;

  export type inject = ((service: IService) => IDecorator) & {
    named: (service: IService, name: string) => IDecorator;
    tagged: (service: IService, name: string, value: any) => IDecorator;
  };

  export type Types = { [key: string]: symbol };

  export const container: Container;
}
