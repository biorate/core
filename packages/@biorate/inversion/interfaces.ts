export namespace IDefine {
  export type Mods = '' | 'c' | 'w' | 'e' | 'cw' | 'we' | 'ce' | 'cwe';
  export type Accessor = { get?(): any; set?(v: any): void };
}
