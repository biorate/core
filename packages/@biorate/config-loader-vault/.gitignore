# compiled output
dist
node_modules
cache
keys

# Logs
logs
*.log
npm-debug.log*
yarn-debug.log*
yarn-error.log*
lerna-debug.log*

# Diagnostic reports (https://nodejs.org/api/report.html)
report.[0-9]*.[0-9]*.[0-9]*.[0-9]*.json

# OS
.DS_Store

# Coverage directory used by tools like istanbul
coverage
*.lcov

# nyc test coverage
.nyc_output

# Optional npm cache directory
.npm

# Optional eslint cache
.eslintcache

# IDEs and editors
.idea
.project
.classpath
.c9/
*.launch
.settings/
*.sublime-workspace

# IDE - VSCode
.vscode/*
!.vscode/settings.json
!.vscode/tasks.json
!.vscode/launch.json
!.vscode/extensions.json

# Package lock
package-lock.json
