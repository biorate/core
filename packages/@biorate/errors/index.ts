import { format } from 'util';

/**
 * @description
 * This module provides a base class for creating error classes with error code (class name),
 * message template, meta data support out of the box.
 *
 * @example
 * ```ts
 * import { BaseError } from '@biorate/errors';
 *
 * export class MyAwesomeError extends BaseError {
 *   constructor(args?: any[], meta?: any) {
 *     super(`Oops... Some error happen, at [%s], in [%s]`, args, meta);
 *   }
 * }
 *
 * const e = new MyAwesomeError([new Date(), 'core'], { hello: 'world!' });
 *
 * console.log(e.meta); // { hello: 'world!' }
 * console.log(e.code); // MyAwesomeError
 *
 * throw e;
 *           ^
 * MyAwesomeError: Oops... Some error happen, at [2021-05-13T09:19:22.511Z], in [core]
 *     at Object.<anonymous> (..core/packages/@biorate/errors/index.ts:28:11)
 *     at Module._compile (internal/modules/cjs/loader.js:1138:30)
 * ```
 */
export class BaseError extends Error {
  #meta: any = null;

  public constructor(
    message: string,
    args?: unknown[],
    meta?: unknown,
    ...options: unknown[]
  ) {
    // @ts-ignore: { cause } property added in 16.9.0
    super(format(message, ...(args || [])), ...options);
    this.#meta = meta;
    this.name = this.constructor.name;
    Error.captureStackTrace(this, this.constructor);
  }

  public get meta() {
    return this.#meta;
  }

  public get code() {
    return this.constructor.name;
  }
}
