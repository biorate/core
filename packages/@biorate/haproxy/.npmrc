registry=https://art.lmru.tech/api/npm/npm-payment/
package-lock=false
