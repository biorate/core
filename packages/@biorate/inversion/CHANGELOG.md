# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.30.2](https://github.com/biorate/core/compare/v0.30.1...v0.30.2) (2022-06-22)

**Note:** Version bump only for package @biorate/inversion





# [0.30.0](https://github.com/biorate/core/compare/v0.29.1...v0.30.0) (2022-06-20)


### Bug Fixes

* *.ts into npmignore ([6a6be03](https://github.com/biorate/core/commit/6a6be030a1e8923df4b03d518f08618e759d06fa))





## [0.29.1](https://github.com/biorate/core/compare/v0.29.0...v0.29.1) (2022-06-20)

**Note:** Version bump only for package @biorate/inversion





# [0.29.0](https://github.com/biorate/core/compare/v0.28.3...v0.29.0) (2022-06-20)


### Bug Fixes

* *.ts into npmignore ([f6e68b3](https://github.com/biorate/core/commit/f6e68b3a2291e7bfa013674db857db7879a26033))





# [0.28.0](https://github.com/biorate/core/compare/v0.27.3...v0.28.0) (2022-05-06)

**Note:** Version bump only for package @biorate/inversion





# [0.27.0](https://github.com/biorate/core/compare/v0.26.0...v0.27.0) (2022-05-06)

**Note:** Version bump only for package @biorate/inversion





# [0.26.0](https://github.com/biorate/core/compare/v0.25.0...v0.26.0) (2022-04-25)

**Note:** Version bump only for package @biorate/inversion





# [0.25.0](https://github.com/biorate/core/compare/v0.24.0...v0.25.0) (2022-04-25)


### Features

* **schema-registry:** api fix ([8631389](https://github.com/biorate/core/commit/8631389036e4c0d1f0a9ec0ccab5a0a2800c4dab))


### BREAKING CHANGES

* **schema-registry:** api fix according axios fatch interface change





# [0.24.0](https://github.com/biorate/core/compare/v0.23.4...v0.24.0) (2022-04-25)

**Note:** Version bump only for package @biorate/inversion





# [0.23.0](https://github.com/biorate/core/compare/v0.22.0...v0.23.0) (2022-04-11)

**Note:** Version bump only for package @biorate/inversion





# [0.22.0](https://github.com/biorate/core/compare/v0.21.0...v0.22.0) (2022-04-08)

**Note:** Version bump only for package @biorate/inversion





# [0.21.0](https://github.com/biorate/core/compare/v0.20.0...v0.21.0) (2022-04-08)

**Note:** Version bump only for package @biorate/inversion





# [0.20.0](https://github.com/biorate/core/compare/v0.19.1...v0.20.0) (2022-04-08)

**Note:** Version bump only for package @biorate/inversion





# [0.19.0](https://github.com/biorate/core/compare/v0.18.2...v0.19.0) (2022-04-07)

**Note:** Version bump only for package @biorate/inversion





## [0.18.2](https://github.com/biorate/core/compare/v0.18.1...v0.18.2) (2022-03-18)


### Bug Fixes

* **inversion:** container multi instantiation remove ([f75e51d](https://github.com/biorate/core/commit/f75e51dea5c617ab4a73c7aeadc1e6eb8dfe9578))





# [0.18.0](https://github.com/biorate/core/compare/v0.17.1...v0.18.0) (2022-02-05)

**Note:** Version bump only for package @biorate/inversion





# [0.17.0](https://github.com/biorate/core/compare/v0.16.3...v0.17.0) (2022-02-04)

**Note:** Version bump only for package @biorate/inversion





## [0.16.3](https://github.com/biorate/core/compare/v0.16.2...v0.16.3) (2022-02-01)

**Note:** Version bump only for package @biorate/inversion





## [0.16.1](https://github.com/biorate/core/compare/v0.16.0...v0.16.1) (2022-02-01)


### Bug Fixes

* publishConfig added in package.json ([716a064](https://github.com/biorate/core/commit/716a06429e1a4325f50cf94b804f1d70d8a58cf5))





# [0.16.0](https://github.com/biorate/core/compare/v0.15.2...v0.16.0) (2022-02-01)

**Note:** Version bump only for package @biorate/inversion





## [0.15.2](https://github.com/biorate/core/compare/v0.15.1...v0.15.2) (2022-01-31)

**Note:** Version bump only for package @biorate/inversion





## [0.15.1](https://github.com/biorate/core/compare/v0.15.0...v0.15.1) (2022-01-31)

**Note:** Version bump only for package @biorate/inversion





# [0.15.0](https://github.com/biorate/core/compare/v0.14.1...v0.15.0) (2022-01-31)


### Features

* **inversion:** all logic related to the life cycle has been moved to the "lifecicled" package ([e747c0c](https://github.com/biorate/core/commit/e747c0ce919ee5cb34a07479b71c0cbdd3c20e4c))





## [0.14.1](https://github.com/biorate/core/compare/v0.14.0...v0.14.1) (2021-12-31)


### Bug Fixes

* **inversion:** Global symbolic registry change ([a185b6f](https://github.com/biorate/core/commit/a185b6f1c746509d3ceea6fe28e36bb96531be5e))





# [0.14.0](https://github.com/biorate/core/compare/v0.13.0...v0.14.0) (2021-12-31)

**Note:** Version bump only for package @biorate/inversion





# [0.13.0](https://github.com/biorate/core/compare/v0.12.4...v0.13.0) (2021-12-31)

**Note:** Version bump only for package @biorate/inversion





## [0.12.2](https://github.com/biorate/core/compare/v0.12.1...v0.12.2) (2021-12-25)


### Bug Fixes

* **all:** homepage and repository in package.json added ([532c5a0](https://github.com/biorate/core/commit/532c5a08ed8cd9c1590134699c074c2021bcbbff))





## [0.12.1](https://github.com/biorate/core/compare/v0.12.0...v0.12.1) (2021-12-25)

**Note:** Version bump only for package @biorate/inversion





# [0.12.0](https://github.com/biorate/core/compare/v0.11.2...v0.12.0) (2021-12-25)

**Note:** Version bump only for package @biorate/inversion





# [0.11.0](https://github.com/biorate/core/compare/v0.10.4...v0.11.0) (2021-10-16)


### Bug Fixes

* **inversion:** cache symbols in global object. Fix multi packages bug ([af7b0a0](https://github.com/biorate/core/commit/af7b0a0dd34512ca815ed0b28760908043f6514a))





## [0.10.4](https://github.com/biorate/core/compare/v0.10.3...v0.10.4) (2021-09-27)

**Note:** Version bump only for package @biorate/inversion





## [0.10.3](https://github.com/biorate/core/compare/v0.10.2...v0.10.3) (2021-09-27)


### Bug Fixes

* **inversion:** Types exports ([4b79327](https://github.com/biorate/core/commit/4b7932724579c5a290e4f5ffa70cd6ad83c2fecc))





## [0.10.2](https://github.com/biorate/core/compare/v0.10.1...v0.10.2) (2021-09-27)

**Note:** Version bump only for package @biorate/inversion





## [0.10.1](https://github.com/biorate/core/compare/v0.10.0...v0.10.1) (2021-09-27)

**Note:** Version bump only for package @biorate/inversion





# [0.10.0](https://github.com/biorate/core/compare/v0.9.3...v0.10.0) (2021-09-24)

**Note:** Version bump only for package @biorate/inversion





## [0.9.3](https://github.com/biorate/core/compare/v0.9.2...v0.9.3) (2021-09-23)


### Bug Fixes

* **inversion:** internal methods names rename ([3f1e3c1](https://github.com/biorate/core/commit/3f1e3c1363bb80e1531793ea7a1c9d772cc0d1e1))





## [0.9.2](https://github.com/biorate/core/compare/v0.9.1...v0.9.2) (2021-09-23)


### Bug Fixes

* **inversion:** exports cleanup ([5124e15](https://github.com/biorate/core/commit/5124e157a223f1549d859c7963a6ce6191657530))





## [0.9.1](https://github.com/biorate/core/compare/v0.9.0...v0.9.1) (2021-09-23)

**Note:** Version bump only for package @biorate/inversion





# [0.9.0](https://github.com/biorate/core/compare/v0.8.4...v0.9.0) (2021-09-23)


### Bug Fixes

* **collection:** build strategy ([40cdf7f](https://github.com/biorate/core/commit/40cdf7f7f62d28a989ce67d94a9125735a0c030d))


### Features

* **inversion:** release ([e479244](https://github.com/biorate/core/commit/e47924417df8f26a2a148d21131ea5c23ac393a0))
* **tools:** env namespace added with isServer value ([69afb82](https://github.com/biorate/core/commit/69afb8273a2319d0423b61b14c7db8f92abce5ab))
