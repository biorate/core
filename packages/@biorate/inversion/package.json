{
  "name": "@biorate/inversion",
  "version": "0.12.1",
  "description": "IoC core module build on InversifyJS",
  "main": "dist",
  "scripts": {
    "build": "npx tsc -p ./tsconfig.build.json --outDir ./dist",
    "test": "npx nyc --reporter=lcov --reporter=text-summary -- npx mocha -r ts-node/register tests/**/*.spec.ts",
    "prepublishOnly": "npm run build"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/biorate/core.git",
    "directory": "packages/@biorate/inversion"
  },
  "homepage": "https://biorate.github.io/core",
  "keywords": [],
  "author": "llevkin",
  "license": "MIT",
  "dependencies": {
    "@biorate/symbolic": "0.12.1",
    "@biorate/tools": "0.12.1",
    "async-exit-hook": "^2.0.1",
    "inversify": "^5.1.1",
    "inversify-inject-decorators": "^3.1.0",
    "lodash": "^4.17.21",
    "reflect-metadata": "^0.1.13"
  },
  "gitHead": "f7398e8c3ab2c828af49f87be9ae58dcfc7c0727"
}
