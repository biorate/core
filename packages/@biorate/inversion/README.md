# Errors factory

This module provides a base class for creating error classes
with **error code** (class name), **message template**, **meta data** support out of the box.

#### Example:
```ts
import { BaseError } from '@biorate/errors';

export class MyAwesomeError extends BaseError {
  constructor(args?: any[], meta?: any) {
    super(`Oops... Some error happen, at [%s], in [%s]`, args, meta);
  }
}

const e = new MyAwesomeError([new Date(), 'core'], { hello: 'world!' });

console.log(e.meta); // { hello: 'world!' }
console.log(e.code); // MyAwesomeError

throw e;
          ^
MyAwesomeError: Oops... Some error happen, at [2021-05-13T09:19:22.511Z], in [core]
    at Object.<anonymous> (..core/packages/@biorate/errors/index.ts:28:11)
    at Module._compile (internal/modules/cjs/loader.js:1138:30)
```

### Learn
* Documentation can be found here - [docs](https://biorate.github.io/core/modules/errors.html).

### Release History
See the [CHANGELOG](https://github.com/biorate/core/blob/master/packages/%40biorate/errors/CHANGELOG.md)

### License
[MIT](https://github.com/biorate/core/blob/master/packages/%40biorate/errors/LICENSE)

Copyright (c) 2021-present [Leonid Levkin (llevkin)](mailto:llevkin@yandex.ru)
