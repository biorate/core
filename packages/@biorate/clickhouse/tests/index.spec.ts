import { expect } from 'chai';
import { container } from '@biorate/inversion';
import { Root } from './__mocks__';

describe('@biorate/clickhouse', function () {
  let root: Root;
  this.timeout(3e4);

  before(async () => {
    root = container.get<Root>(Root);
    await root.$run();
  });

  it('test', async () => {
    const data = await root.connector!.query<{ result: number }>('SELECT 1 AS result;');
    expect(data[0].result).toMatchSnapshot();
  });
});
