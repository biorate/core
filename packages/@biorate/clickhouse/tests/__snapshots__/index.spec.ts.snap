// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@biorate/clickhouse test 1`] = `1`;
